### PR TITLE
Refact state-widget component

### PR DESCRIPTION
**Pull request by GabrielVidal1**
Refact the state-widget component used in the chapter children list.

### CODE REVIEW

File: src/app/modules/shared-components/components/state-widget/state-widget.component.ts [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
 
-  labels = ['Discovery', 'Practice', 'Validation', 'Challenge', 'Review'];
+  @Input() type: 'Undefined'|'Discovery'|'Application'|'Validation'|'Challenge';
+  @Input() disabled = false;
 
-  classes = ['discovery', 'practice', 'validation', 'challenge', 'review'];
+  icon = '';
 
-  constructor() {}
+  ngOnInit(): void {
```
Comment (smadbe) [inline src/app/modules/shared-components/components/state-widget/state-widget.component.ts:15]: on change

File: src/app/modules/shared-components/components/state-widget/state-widget.component.scss [hunk 2]
```diff
@@ -8,7 +8,7 @@
   display: flex;
   font-size: .9rem;
 
-  &.discovery {
+  &.Undefined {
```
Comment (smadbe) [inline src/app/modules/shared-components/components/state-widget/state-widget.component.scss:11]: Use a stored variable which is the lowercase of type instead.

**Review by smadbe — CHANGES_REQUESTED** [review]

File: src/app/modules/shared-components/components/state-widget/state-widget.component.ts (hunk 1 above)
Comment (GabrielVidal1) [reply src/app/modules/shared-components/components/state-widget/state-widget.component.ts]: fixed

File: src/app/modules/shared-components/components/state-widget/state-widget.component.scss (hunk 2 above)
Comment (GabrielVidal1) [reply src/app/modules/shared-components/components/state-widget/state-widget.component.scss]: fixed

**Review by smadbe — APPROVED** [review]

**Pull request merged**